### PR TITLE
fix(grpc): no package in method service

### LINKED ISSUE #2375: grpc client fails when parse_method_path() does not return a value for package
### Which version of dd-trace-py are you using?

ddtrace==0.48.0

### Which version of pip are you using?

pip 21.1

### Which version of the libraries are you using?

ddtrace==0.48.0

### How can we reproduce your problem?

parse_method_path() can [return None](https://github.com/DataDog/dd-trace-py/blob/7f817089fac3efc2d1204d347e5d21abe344445c/ddtrace/contrib/grpc/utils.py#L12) for the `package` value. This is then passed to [span._set_str_tag()](https://github.com/DataDog/dd-trace-py/blob/7f817089fac3efc2d1204d347e5d21abe344445c/ddtrace/contrib/grpc/client_interceptor.py#L184), which expects the value to always be `str`.

### What is the result that you get?
```2021-04-29 12:50:27 [6] [ERROR] grpc._server     Exception calling application: not expecting type '<class 'NoneType'>'
Traceback (most recent call last):
  File "/pyenv/versions/app_venv/lib/python3.7/site-packages/grpc/_server.py", line 435, in _call_behavior
    response_or_iterator = behavior(argument, context)
  File "/pyenv/versions/app_venv/lib/python3.7/site-packages/ddtrace/contrib/grpc/server_interceptor.py", line 112, in unary_unary
    return self._fn(constants.GRPC_METHOD_KIND_UNARY, self.__wrapped__.unary_unary, args, kwargs)
  File "/pyenv/versions/app_venv/lib/python3.7/site-packages/ddtrace/contrib/grpc/server_interceptor.py", line 79, in _fn
    span._set_str_tag(constants.GRPC_METHOD_PACKAGE_KEY, method_package)
  File "/pyenv/versions/app_venv/lib/python3.7/site-packages/ddtrace/span.py", line 318, in _set_str_tag
    self.meta[key] = ensure_text(value, errors="replace")
  File "/pyenv/versions/app_venv/lib/python3.7/site-packages/ddtrace/vendor/six/__init__.py", line 937, in ensure_text
    raise TypeError("not expecting type '%s'" % type(s))
TypeError: not expecting type '<class 'NoneType'>'
```

### What is the result that you expected?
The interceptor does not raise exceptions when the value is not available.

The issue seems to be surfaces as a result of #2218. it's not clear if the method package should be omitted or converted to a string as done in some of the other integrations edited in that PR.

### PR DESCRIPTION
**Pull request by majorgreys**
## Description

Fixes #2375 

Failing test run with snapshot test: https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/7778/workflows/fce7f1aa-9965-491d-935e-1486e0f3e21c/jobs/657191

## Checklist
- [ ] Added to the correct milestone.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by P403n1x87 — APPROVED** [review]

File: tests/contrib/grpc/test_grpc.py [hunk 1]
```diff
@@ -572,3 +574,40 @@ class NotFutureLike(object):
...
+
+class _UnaryUnaryRpcHandler(grpc.GenericRpcHandler):
+    def __init__(self, handler):
+        self._handler = handler
+
+    def service(self, handler_call_details):
+        return grpc.unary_unary_rpc_method_handler(self._handler)
+
+
+@snapshot(ignores=["meta.grpc.port"])
```
Comment (Kyle-Verhoog) [inline tests/contrib/grpc/test_grpc.py:596]: This snapshot could also catch the server spans if we wanted with an `async_mode=False`.

@majorgreys should we also capture the server span?

Comment (majorgreys) [reply tests/contrib/grpc/test_grpc.py:596]: The server and client, in this run at least, share the same global tracer (through the pin) running in the same process. WDYT?

Comment (Kyle-Verhoog) [reply tests/contrib/grpc/test_grpc.py:596]: Ohh the server span is there! I missed it because of the matching operation name

**Review by Kyle-Verhoog — APPROVED** [review]

**Comment by Kyle-Verhoog** [comment]
@mergifyio backport 0.48

**Pull request merged**